From 066ad58e7e0c0ad89954cb16220c53db4a1cada8 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:05:37 -0400
Subject: [usb] uhci-platform: Change compatible string from platform-uhci to generic-uhci

Message-id: <1410887463-20674-266-git-send-email-dzickus@redhat.com>
Patchwork-id: 95235
O-Subject: [RHEL7 PATCH 265/591] uhci-platform: Change compatible string from platform-uhci to generic-uhci
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit e16fa44b39189cf31baaa8e880bc1c23a458c669
Author: Hans de Goede <hdegoede@redhat.com>
Date:   Tue Feb 11 17:54:45 2014 +0100

    uhci-platform: Change compatible string from platform-uhci to generic-uhci

    This brings the uhci-platform bindings in sync with what we've done for
    the ohci- and ehci-platform drivers. As discussed there using platform as a
    prefix is a bit weird as the platform bus is a Linux specific thing and
    the bindings are supposed to be OS agnostic.

    Note that the old platform-uhci compatible string is kept around for, well,
    compatibility reasons.

    While at it rename the bindings txt file to match the name of all the
    other ?hci-platform bindings docs.

    Signed-off-by: Hans de Goede <hdegoede@redhat.com>
    Acked-by: Alan Stern <stern@rowland.harvard.edu>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---
 delete mode 100644 Documentation/devicetree/bindings/usb/platform-uhci.txt
 create mode 100644 Documentation/devicetree/bindings/usb/usb-uhci.txt

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/Documentation/devicetree/bindings/usb/platform-uhci.txt b/Documentation/devicetree/bindings/usb/platform-uhci.txt
deleted file mode 100644
index a4fb071..0000000
--- a/Documentation/devicetree/bindings/usb/platform-uhci.txt
+++ /dev/null
@@ -1,15 +0,0 @@
-Generic Platform UHCI Controller
------------------------------------------------------
-
-Required properties:
-- compatible : "platform-uhci"
-- reg : Should contain 1 register ranges(address and length)
-- interrupts : UHCI controller interrupt
-
-Example:
-
- uhci@d8007b00 {
-  compatible = "platform-uhci";
-  reg = <0xd8007b00 0x200>;
-  interrupts = <43>;
- };
diff --git a/Documentation/devicetree/bindings/usb/usb-uhci.txt b/Documentation/devicetree/bindings/usb/usb-uhci.txt
new file mode 100644
index 0000000..2981334
--- /dev/null
+++ b/Documentation/devicetree/bindings/usb/usb-uhci.txt
@@ -0,0 +1,15 @@
+Generic Platform UHCI Controller
+-----------------------------------------------------
+
+Required properties:
+- compatible : "generic-uhci" (deprecated: "platform-uhci")
+- reg : Should contain 1 register ranges(address and length)
+- interrupts : UHCI controller interrupt
+
+Example:
+
+ uhci@d8007b00 {
+  compatible = "generic-uhci";
+  reg = <0xd8007b00 0x200>;
+  interrupts = <43>;
+ };
diff --git a/drivers/usb/host/uhci-platform.c b/drivers/usb/host/uhci-platform.c
index 50d2f7f..8486595 100644
--- a/drivers/usb/host/uhci-platform.c
+++ b/drivers/usb/host/uhci-platform.c
@@ -149,6 +149,7 @@ static void uhci_hcd_platform_shutdown(struct platform_device *op)
 }
 
 static const struct of_device_id platform_uhci_ids[] = {
+ { .compatible = "generic-uhci", },
  { .compatible = "platform-uhci", },
  {}
 };
-- 
1.7.1